config/config_test.go

```diff
@@ -0,0 +1,333 @@
+package config
+
+import (
+	"crypto/tls"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/consul-ecs/awsutil"
+	"github.com/hashicorp/consul-ecs/testutil"
+	"github.com/hashicorp/consul-server-connection-manager/discovery"
+	"github.com/hashicorp/consul/api"
+	"github.com/stretchr/testify/require"
+)
+
+const testCA = `
+-----BEGIN CERTIFICATE-----
+MIIC7TCCApOgAwIBAgIQbHoocPoQq7qR3MTNUXdLVDAKBggqhkjOPQQDAjCBuTEL
+MAkGA1UEBhMCVVMxCzAJBgNVBAgTAkNBMRYwFAYDVQQHEw1TYW4gRnJhbmNpc2Nv
+MRowGAYDVQQJExExMDEgU2Vjb25kIFN0cmVldDEOMAwGA1UEERMFOTQxMDUxFzAV
+BgNVBAoTDkhhc2hpQ29ycCBJbmMuMUAwPgYDVQQDEzdDb25zdWwgQWdlbnQgQ0Eg
+MTQ0MTkwOTA0MDA4ODQxOTE3MTQzNDM4MjEzMTEzMjA0NjU2OTgwMB4XDTIyMDkx
+NjE4NDUwNloXDTI3MDkxNTE4NDUwNlowgbkxCzAJBgNVBAYTAlVTMQswCQYDVQQI
+EwJDQTEWMBQGA1UEBxMNU2FuIEZyYW5jaXNjbzEaMBgGA1UECRMRMTAxIFNlY29u
+ZCBTdHJlZXQxDjAMBgNVBBETBTk0MTA1MRcwFQYDVQQKEw5IYXNoaUNvcnAgSW5j
+LjFAMD4GA1UEAxM3Q29uc3VsIEFnZW50IENBIDE0NDE5MDkwNDAwODg0MTkxNzE0
+MzQzODIxMzExMzIwNDY1Njk4MDBZMBMGByqGSM49AgEGCCqGSM49AwEHA0IABA9w
+J9aqbpdoVXQLdYTfUpBM2bgElznRYQP/GcNQUtvopvVywPjC7obFuZP1oM7YX7Wy
+hGyeudV4pvF1lz9nVeOjezB5MA4GA1UdDwEB/wQEAwIBhjAPBgNVHRMBAf8EBTAD
+AQH/MCkGA1UdDgQiBCA9dZuoEX3yrbebyEEzsN4L2rr7FJd6FsjIioR6KbMIhTAr
+BgNVHSMEJDAigCA9dZuoEX3yrbebyEEzsN4L2rr7FJd6FsjIioR6KbMIhTAKBggq
+hkjOPQQDAgNIADBFAiARhJR88w9EXLsq5A932auHvLFAw+uQ0a2TLSaJF54fyAIh
+APQczkCoIFiLlGp0GYeHEfjvrdm2g8Q3BUDjeAUfZPaW
+-----END CERTIFICATE-----`
+
+func TestConsulServerConnManagerConfig(t *testing.T) {
+	cases := map[string]struct {
+		cfg       *Config
+		taskMeta  awsutil.ECSTaskMeta
+		expConfig func(awsutil.ECSTaskMeta) discovery.Config
+	}{
+		"basic flags without TLS or ACLs": {
+			cfg: &Config{
+				ConsulServers: ConsulServers{
+					Hosts:    "consul.dc1.address",
+					GRPCPort: 8503,
+				},
+			},
+			expConfig: func(t awsutil.ECSTaskMeta) discovery.Config {
+				return discovery.Config{
+					Addresses: "consul.dc1.address",
+					GRPCPort:  8503,
+				}
+			},
+		},
+		"basic flags without TLS or ACLs with custom gRPC port": {
+			cfg: &Config{
+				ConsulServers: ConsulServers{
+					Hosts:    "consul.dc1.address",
+					GRPCPort: 8502,
+				},
+			},
+			expConfig: func(t awsutil.ECSTaskMeta) discovery.Config {
+				return discovery.Config{
+					Addresses: "consul.dc1.address",
+					GRPCPort:  8502,
+				}
+			},
+		},
+		"default TLS": {
```

Comment:
nit: naming - because we aren't testing the config parsing/validation here, these tests aren't validating what default values are used, so it makes sense to just name these tests based on which settings are enabled

```suggestion
		"TLS enabled": {
```